integration_test/ginkgo_integration_test.go

```diff
@@ -0,0 +1,61 @@
+package integrationtest
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/node-disk-manager/integration_test/minikube_adm"
+	"github.com/openebs/node-disk-manager/integration_test/ndm_util"
+)
+
+func TestIntegrationNDM(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Integration Test Suit")
+}
+
+var _ = BeforeSuite(func() {
+	// It starts minikube if it is not Running
+	minikubeadm.Setup()
+
+	// It waits till namespace is ready
+	ndmutil.WaitTillDefaultNSisReady()
+
+	// It applies our CRDs
+	ndmutil.ApplyCRDs()
+
+	// It prepares configuration and Applies the same
+	ndmutil.PrepareAndApplyYAML()
+
+	// It waits till node-disk-manager is ready
+	ndmutil.WaitTillNDMisUp()
+})
+
+var _ = AfterSuite(func() {
+	// It Delete minikube if it is running
+	// It removes residue containers
+	// It removes remaining residue files
+	ndmutil.Clean()
+})
+
+var _ = Describe("Integration Test", func() {
+	// Now as BeforeSuit has run, We shall have a healthy node-disk-manager daemonset
+	When("We check the log", func() {
+		It("has `Started the controller` in the log", func() {
+			validated, err := ndmutil.GetNDMLogAndValidate()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(validated).To(BeTrue())
```

Comment:
It seems it is very very essential to have variable names which can get along with ginkgo & gomega methods & in turn the entire method invocation can be read as a sentence.

e.g. `expect err not to have occurred` is coded as `Expect(err).NotTo(HaveOccurred())`
Lets try to keep above in mind when we name the methods & variables.

Reply:
I tried to follow the same, is there any thing that I missed? should I rename variable `validated` to `validation` or something else?

Reply:
ofcourse you have followed it... just maintain it.

Reply:
Sure.